### PR TITLE
[discussing] Add mypy_extensions.

### PR DESCRIPTION
**Pull request by davidfstr**
Supports pending PR https://github.com/python/mypy/pull/2206 .

### CODE REVIEW

File: third_party/2and3/mypy_extensions.pyi [hunk 1]
```diff
@@ -0,0 +1,6 @@
+from typing import Dict, Type, TypeVar
+
+T = TypeVar('T')
+
+
+def TypedDict(typename: str, fields: Dict[str, Type[T]]) -> Type[dict]: ...
```
Comment (gvanrossum) [inline third_party/2and3/mypy_extensions.pyi]: Hm, I thought the design for TypeDict allowed it to be used with keyword args?

Comment (davidfstr) [inline third_party/2and3/mypy_extensions.pyi]: It's not supported in the current implementation so far, but yes it's in the design.

I have a line item pending to discuss some forward compatibility concerns with keyword args (`**kwfields`) before actually implementing it.

**Comment by gvanrossum** [comment]
OK, let's wait until we've discussed that then.

**Comment by gvanrossum** [comment]
Actually it's better to merge this before merging https://github.com/python/mypy/pull/2206, so I've done that now.

**Pull request merged**